pkg/kubelet/status/manager.go

```diff
@@ -382,6 +382,11 @@ func (m *manager) syncPod(uid types.UID, status versionedPodStatus) {
 				glog.V(3).Infof("Pod %q is terminated, but some containers are still running", format.Pod(pod))
 				return
 			}
+			if pod.DeletionGracePeriodSeconds != nil && *pod.DeletionGracePeriodSeconds == 0 {
```

Comment:
If it's kubelet's responsibility to ensure that a pod is removed from etcd, there is bound to be races where kubelet deletes the object before the apiserver, which is what @caesarxuchao wanted to avoid to begin with.